libcoap-sys/build.rs

```diff
@@ -24,25 +59,383 @@ pub enum DtlsBackend {
     MbedTls,
     TinyDtls,
 }
-impl ToString for DtlsBackend {
-    fn to_string(&self) -> String {
-        match self {
+impl Display for DtlsBackend {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let str = match self {
             DtlsBackend::GnuTls => "gnutls",
             DtlsBackend::OpenSsl => "openssl",
             DtlsBackend::MbedTls => "mbedtls",
             DtlsBackend::TinyDtls => "tinydtls",
         }
-        .to_string()
+        .to_string();
+        write!(f, "{}", str)
+    }
+}
+
+fn get_target_mcu() -> &'static str {
+    let cfg_flags = embuild::espidf::sysenv::cfg_args().expect("missing cfg flags from IDF");
+    let mcus = [
+        "esp32", "esp32s2", "esp32s3", "esp32c3", "esp32c2", "esp32h2", "esp32c5", "esp32c6", "esp32p4",
+    ];
+    for mcu in mcus {
+        if cfg_flags.get(mcu).is_some() {
+            return mcu;
+        }
+    }
+    panic!("unknown ESP target MCU, please add target to libcoap-sys build.rs file!")
+}
+
+fn get_builder_espidf() -> bindgen::Builder {
+    embuild::espidf::sysenv::output();
+    let esp_idf_path = embuild::espidf::sysenv::idf_path().expect("missing IDF path");
+    let esp_idf_buildroot = env::var("DEP_ESP_IDF_ROOT").expect("DEP_ESP_IDF_ROOT is not set");
+    let esp_include_path = embuild::espidf::sysenv::cincl_args().expect("missing IDF cincl args");
+    let embuild_env = embuild::espidf::sysenv::env_path().expect("missing IDF env path");
+    let esp_arch = env::var("CARGO_CFG_TARGET_ARCH").expect("CARGO_CFG_TARGET_ARCH is not set");
+
+    // Determine compiler path
+    unsafe { env::set_var("PATH", embuild_env) };
```

Comment:
There is one warning by the linter left here that indicates that the `unsafe` block is not necessary.

```suggestion
    env::set_var("PATH", embuild_env);
```

Reply:
Starting with more recent stable rust versions, this function call is declared as unsafe.
See: https://doc.rust-lang.org/std/env/fn.set_var.html#safety and https://github.com/rust-lang/rust/issues/27970

Reply:
Thanks, makes sense to keep it, then :) Would it make sense to add an allow annotation here to indicate that this intended?